examples/use_case_streaming_agents/react/backend.py

```diff
@@ -0,0 +1,57 @@
+from dynamiq.callbacks.streaming import StreamingIteratorCallbackHandler
+from dynamiq.connections import E2B, ScaleSerp
+from dynamiq.memory import Memory
+from dynamiq.memory.backend.in_memory import InMemory
+from dynamiq.nodes.agents.react import ReActAgent
+from dynamiq.nodes.tools.e2b_sandbox import E2BInterpreterTool
+from dynamiq.nodes.tools.scale_serp import ScaleSerpTool
+from dynamiq.runnables import RunnableConfig
+from dynamiq.types.streaming import StreamingConfig, StreamingMode
+from examples.llm_setup import setup_llm
+
+
+def setup_agent(agent_role: str, streaming_enabled: bool, streaming_mode: str) -> ReActAgent:
+    """
+    Initializes an AI agent with a specified role and streaming configuration.
+    """
+    llm = setup_llm()
+    memory = Memory(backend=InMemory())
+    mode_mapping = {"Final": StreamingMode.FINAL, "All": StreamingMode.ALL}
+    mode = mode_mapping.get(streaming_mode, StreamingMode.FINAL)
+    streaming_config = StreamingConfig(enabled=streaming_enabled, mode=mode)
+    tool_search = ScaleSerpTool(connection=ScaleSerp())
+    tool_code = E2BInterpreterTool(connection=E2B())
+    agent = ReActAgent(
+        name="Agent",
+        llm=llm,
+        role=agent_role,
+        id="agent",
+        memory=memory,
+        tools=[tool_code, tool_search],
+        streaming=streaming_config,
+    )
+    return agent
+
+
+def generate_agent_response(agent: ReActAgent, user_input: str):
+    """
+    Processes the user input using the agent. Supports both streaming and non-streaming responses.
+    """
+    if agent.streaming.enabled:
+        streaming_handler = StreamingIteratorCallbackHandler()
+        agent.run(
```

Comment:
To unblock this operation, don't wait result of it and start streaming immediately you could run it in separate thhread. Im similar way we managed streaming in async:
https://github.com/dynamiq-ai/dynamiq/blob/main/examples/chainlit/chainlit_agent.py#L76
https://github.com/dynamiq-ai/dynamiq/blob/main/examples/chainlit/chainlit_agent.py#L26